src/github.com/getlantern/lantern-android/client/client.go

```diff
@@ -110,18 +104,11 @@ func (client *MobileClient) ServeHTTP() {
 // updateConfig attempts to pull a configuration file from the network using
 // the client proxy itself.
 func (client *MobileClient) updateConfig() error {
-	var err error
 	var buf []byte
-	var cli *http.Client
-
-	if cli, err = util.HTTPClient(cloudConfigCA, client.Addr); err != nil {
-		return err
-	}
```

Comment:
Not as bad, but close.  Note that

- The default configuration [didn't include](https://github.com/getlantern/lantern/pull/2538/files#diff-8be20108b452119099e317b2a3b4e03aR116) the trusted CAs.  This was probably a bug and not by design.

- The client was initially [trying](https://github.com/getlantern/lantern/pull/2538/files#diff-8be20108b452119099e317b2a3b4e03aL131) to fetch a new config without proxying.  Only if this succeeds will the trusted CAs be properly initialized.

BTW, I just realised that the trusted CAs were never updated after initial setup.  I just submitted a fix.